src/app/TimedHandler.cpp

```diff
@@ -66,10 +66,9 @@ CHIP_ERROR TimedHandler::OnMessageReceived(Messaging::ExchangeContext * aExchang
                       ChipLogValueX64(now.count()), this, ChipLogValueExchange(aExchangeContext));
         if (now > mTimeLimit)
         {
-            // Time is up.  Spec says to send UNSUPPORTED_ACCESS.
             ChipLogError(DataManagement, "Timeout expired: handler %p exchange " ChipLogFormatExchange, this,
                          ChipLogValueExchange(aExchangeContext));
-            return StatusResponse::Send(Status::UnsupportedAccess, aExchangeContext, /* aExpectResponse = */ false);
+            return StatusResponse::Send(Status::Timeout, aExchangeContext, /* aExpectResponse = */ false);
```

Comment:
Looking at https://github.com/CHIP-Specifications/connectedhomeip-spec/commit/0bd0e4f937ebaf84d6ebe7dc22e79c5af605bc2a and the current spec text we also need to fix `WriteHandler::ProcessWriteRequest`.  The `mIsTimedRequest != aIsTimedWrite` bit needs to return TimedRequestMismatch, right?  It looks like this was changed for invoke but not write, probably because we don't have any actual timed writable attributes to test on....

Reply:
If it's ok, I'd like to address this in a followup. We can do a non-cert test of a writable timed attribute using the unit testing cluster, but I don't believe I will have a change to add attributes and tests etc. before the TE2 deadline. I can make an issue if the write is not a blocker for this PR.

Reply:
As it turns out, we have a timed write attribute in the unit testing cluster and a test, so may as well do this now.

Reply:
Actually, it turns out we DO have a unit test for this, so I'm more comfortable just fixing this here. Opened an issue to add to cert as that's going to require a LOT more plumbing. Issue here: https://github.com/CHIP-Specifications/chip-test-plans/issues/4256